core/src/com/unciv/ui/tilegroups/CityButton.kt

```diff
@@ -70,13 +70,14 @@ class CityButton(val city: CityInfo, internal val tileGroup: WorldTileGroup, ski
             // clicking swings the button a little down to allow selection of units there.
             // this also allows to target selected units to move to the city tile from elsewhere.
             if (isButtonMoved) {
+                val currentCiv = UncivGame.Current.worldScreen.viewingCiv
                 // second tap on the button will go to the city screen
                 // if this city belongs to you
                 if (UncivGame.Current.viewEntireMapForDebug || belongsToViewingCiv()) {
                     UncivGame.Current.setScreen(CityScreen(city))
-                } else {
+                } else if (currentCiv.knows(city.civInfo)) {
```

Comment:
This should probably not reference the currentCiv. In Multiplayer, currentCiv is the civ who's turn it is, but if you're a different player then you should get to 'your own' diplomacy screen, and not that of the current player

Reply:
As you see a few lines higher, it uses the `viewingCiv`
> val currentCiv = UncivGame.Current.worldScreen.viewingCiv
Is it a right way to get 'your own' civilization?

Reply:
I see, then it's functionally perfect =)
Even so, please change the parameter's name to viewingCiv, because currentCiv already has a different connotation

Reply:
I agree, the naming caused the confusion, will rephrase.